### ISSUE TITLE
[BUG] hyprland 2.9.7.0  crashing after todays update 

### ISSUE THREAD

**Issue by Koushikg88** [mylinuxforwork/dotfiles#515]
**Describe the bug**
after doing system update if rebooted, hyprland is not openning and sending bck to login screen

**To Reproduce**
1. get latest update
2. reboot the system
3. loging to hyprland

**Expected behavior**
A clear 

**Distribution (please complete the following information):**
 - Arch Linux

**Comment by Koushikg88** [comment]
[hyprlandCrashReport28295.txt](https://github.com/user-attachments/files/18042432/hyprlandCrashReport28295.txt)
thiss is the crash report

**Comment by Haraven** [comment]
**Tl; dr**: If you have this line in your Hyprland config:
`env = LIBGL_ALWAYS_SOFTWARE,1`
try removing it.

In ML4W, it's a configuration option found under the `~/.config/hypr/conf/environments/kvm.conf`. If you're using the KVM environment preset or parts taken from it, that might be the culprit.

---

I had the same issue: after updating, I could no longer boot into Hyprland.

I found no real solutions online, but a Reddit post mentioned trying to run Hyprland without a config to see if it works that way. For me, running without a config meant commenting everything in `~/.config/hypr/hyprland.conf`.

I did that, and surprisingly, Hyprland launched. So, what I did was to go back to `~/.config/hypr/hyprland.conf` and to uncomment the configs, **line by line**, until I found the issue. In my case, the issue was caused by an environment variable from the KVM preset that ML4W comes with (`~/.config/hypr/conf/environments/kvm.conf`). The line causing the crash was:
`env = LIBGL_ALWAYS_SOFTWARE,1`

I commented that line and Hyprland launched just fine after that.

**Comment by Koushikg88** [comment]
Yes, thats right, I did the same you did and its working for me too. Thanks for help.

**Comment by mylinuxforwork** [comment]
@Haraven @Koushikg88 Thanks for testing. So, you’re running the dotfiles in KVM and you removed this environment variable and then it works again?

**Comment by Koushikg88** [comment]
> @Haraven @Koushikg88 Thanks for testing. So, you’re running the dotfiles in KVM and you removed this environment variable and then it works again?

Yes, I have commented that particular line in kvm.conf inside environments folder and its working fine.

**Comment by Haraven** [comment]
> @Haraven @Koushikg88 Thanks for testing. So, you’re running the dotfiles in KVM and you removed this environment variable and then it works again?

I have ML4W installed on bare metal. I was using the environment variables for KVM, under the assumption that it helps with performance when I need to run my W11 VM inside Hyprland.

Are we supposed to only use the KVM variables when we're running ML4W inside a VM? :thinking:

**Comment by degobbis** [comment]
Nice Info!
Solved my issue on Hyprland: https://github.com/hyprwm/Hyprland/issues/8666

**Comment by mylinuxforwork** [comment]
> > @Haraven @Koushikg88 Thanks for testing. So, you’re running the dotfiles in KVM and you removed this environment variable and then it works again?
> 
> I have ML4W installed on bare metal. I was using the environment variables from `kvm.conf`, under the assumption that it helps with performance when I need to run my W11 VM inside Hyprland. When I removed the environment variable, Hyprland started working again.
> 
> Are we supposed to only use the KVM variables when we're running ML4W inside a VM? :thinking: 

Yesm The KVM environment should only be used in a VM.
